### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by Christabelle143**
# In this project i completed the following tasks;

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

**Comment by Christabelle143** [comment]
Hey, 
This is for the reviewer. Thanks for taking time to review my project. I lack one thing on the project which is the live link. I have been working on it for like 3 hours now to no avail using render. Tried using Heroku but can't proceed since I don't have a credit card. Will work on it further but due to the deadline I have to submit my project without it for now. I will definitely add it when I succeed. Hope you will be compassionate with me for this  week reviewer.
Thanks in advance.
Here is a screenshots of my different trials on render
render

**Review by rloterh — COMMENTED** [review]
Hi Christabelle, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

I get the error below in attempt to start your application. Kindly ensure to update the online version of your code if your local version of code is working as intended:

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Christabelle143** [comment]
just fixed the issue thanks.

File: .stylelintrc.json [hunk 1]
```diff
@@ -2,8 +2,30 @@
...
+          "tailwind",
+          "apply",
+          "variants",
+          "responsive",
+          "screen"
+        ]
+      }
+    ],
     "csstree/validator": true
   },
```
Comment (andersonlebon) [inline .stylelintrc.json:30]: - [ ] I have Tested your project locally and it's really nice, nice job on using the active storage to attach to the transaction a logo or avatar(jpg, png...), However, your project still needs to follow the styling requirements of the project
image image image image image 

The images above show all the Design requirements you should follow:
image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Christabelle143 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

almost-there

## Suggested changes
_Check the comments under the review._

- [ ] Nice work on recording a looming video :clap: :clap:  however, one of the video presentation requirements says that your video should contain a highlight about a piece of the code that you find interesting or you are proud of making. Kindly work on this so that your project gets strong documentation.

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Lameck1 — APPROVED** [review]
Hi, @Christabelle143,

**_This is a manual review._**

Tasks for this project have been successfully implemented 💯.
You understood the assignment and have done a good job 👍🏼.

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Pull request merged**